index.js

```diff
@@ -26,42 +24,83 @@ const test = (whatWeTest, actualResult, expectedResult) => {
 
 // Functions
 
-const getType = (value) => {
-    // Return string with a native JS type of value
-};
+const getType = (value) => typeof value;
 
-const getTypesOfItems = (arr) => {
-    // Return array with types of items of given array
-};
+const getTypesOfItems = (arr) => arr.map((elmnt) => typeof elmnt);
 
 const allItemsHaveTheSameType = (arr) => {
-    // Return true if all items of array have the same type
+    if (arr.length > 1) {
+        for (let i = 1; i < arr.length; i++) {
+            if (typeof arr[0] !== typeof arr[i]) {
+                return false;
+            }
+        }
+    }
+    return true;
 };
 
 const getRealType = (value) => {
-    // Return string with a “real” type of value.
-    // For example:
-    //     typeof new Date()       // 'object'
-    //     getRealType(new Date()) // 'date'
-    //     typeof NaN              // 'number'
-    //     getRealType(NaN)        // 'NaN'
-    // Use typeof, instanceof and some magic. It's enough to have
-    // 12-13 unique types but you can find out in JS even more :)
-};
+    switch (true) {
+        case value instanceof Array:
+            return 'array';
+
+        case value instanceof Date:
+            return 'date';
+
+        case value instanceof RegExp:
+            return 'regexp';
+
+        case value instanceof Set:
+            return 'set';
+
+        case Number.isNaN(value):
+            return 'NaN';
 
-const getRealTypesOfItems = (arr) => {
-    // Return array with real types of items of given array
+        case value === Number.POSITIVE_INFINITY:
+            return 'Infinity';
+
+        case value === null:
+            return 'null';
+
+        default:
+            return typeof value;
+    }
 };
 
+const getRealTypesOfItems = (arr) => arr.map((elmnt) => getRealType(elmnt));
+
 const everyItemHasAUniqueRealType = (arr) => {
-    // Return true if there are no items in array
-    // with the same real type
+    const realTypesArray = getRealTypesOfItems(arr);
+
+    for (let i = 0; i < realTypesArray.length - 1; i++) {
+        for (let j = i + 1; j < realTypesArray.length; j++) {
+            if (realTypesArray[j] === realTypesArray[i]) {
+                return false;
+            }
+        }
+    }
```

Comment:
```suggestion
    while (realTypesArray.length > 1) {
        const firstArrayItem = realTypesArray.shift();
        if (realTypesArray.some((arrayElement) => arrayElement === firstArrayItem)) {
            return false;
        }
    }
```
Есть такой вариант, но по сути это просто махинации с синтаксисом, с точки зрения использования ресурсов те же грабли, так что пока думаю))